### ISSUE TITLE
Blocked contacts still able to receive public posts and comment on their node

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#10049]
This is less a bug report than an unpleasant experience report. I've been able to find on their node a comment on one of my public posts from a contact I blocked.

After the unpleasant surprise, I realized that, as a public post, it is impossible to limit who can have access to it, but the fact that I was surprised feels like we probably aren't explaining the negative interaction between blocking a contact and public posts correctly.

I don't know where to put this particular bit of information, maybe in the permission screen in the "Public" box?

**Comment by annando** [comment]
One could write a full chapter about this topic I guess. Of course we could place it in the documentation, but question is how many would read this. And I'm not sure if some brief text can explain the problem in an understandable way.

**Comment by tio-trom** [comment]
Shouldn't posts act as "master" and all comments as "slaves" - like with hard drives. So that the post commands what comments to be displayed or not, and update that information to all servers. 

Example:

If Aaron makes a post, the post is then sent to multiple servers on multiple federated networks. Now Aaron deletes the post, the information should be sent to all servers and the post should be removed. In the same manner, if Aaron deletes a comment on his post, then the information should be sent to all servers and the comment should be deleted from all networks. Lastly, if Aaron blocks person X, the person should not be able to comment anymore on his posts or interact with them, and the information should be sent to all servers - meaning: if the blocked person comments on Aaron's post on his own node or any other federated network, then the comment should wait until the original post either approves it or not.

I think that's how it should work....

**Comment by MrPetovan** [comment]
These are the expectations in a centralized system, and they unfortunately can't apply as well to decentralized systems. In the latter, you only control your end of the system, which means that you can only block incoming content or sign outgoing content against a particular remote contact cryptographic key (limited/private posts). But public content is public, and you can't prevent remote contacts to interact with one of your public post on their own node.

Friendica is in a weird spot where it looks and feels like Facebook where posts are limited to friends by default if I remember correctly, and we're also interfacing with Twitter-like networks like Mastodon where posts are public by default. But even Twitter's concept of a "public post" is more limited than an ActivityPub public post, so there are a lot of expectation to unlearn moving from centralized to decentralized social media.

And if I could get unpleasantly surprised 4 years into my own switch, actively working on the Friendica project and hosting my own single-user node, I can only imagine what it must be like for more regular users.

**Comment by tio-trom** [comment]
Interesting indeed. So it means that if one makes a public post with something embarrassing they can never delete that post from the fediverse, only from their own node. Then, I wonder, what does the "delete globally" option on Friendica is about?

**Comment by AlfredSK** [comment]
Not quite. We are sending 'delete' messages to other systems. But it depends on the remote system if the 'delete' is processed. From my experience it works. E.g. when you delete one of your own posts it is also removed on Mastodon.

**Comment by tio-trom** [comment]
> Not quite. We are sending 'delete' messages to other systems. But it depends on the remote system if the 'delete' is processed. From my experience it works. E.g. when you delete one of your own posts it is also removed on Mastodon.

This makes sense. So, wouldn't it work to apply the same practice for comments? Like I said above, in the same way that if you delete a post of your own, it sends a "delete" message to the other networks, what if when you delete a comment from your own post, it would send the same request to the other servers? And then, to make it better, if someone is blocked by you, and that someone comments on your post (regardless where - on your network or other networks), the comment to be deleted from every network in the same fashion.

**Comment by AlfredSK** [comment]
It doesn't matter if it is a top level post (parent) or comment (child). It works basically the same way.

**Comment by MrPetovan** [comment]
Deleting your own content and deleting someone else's content is completely different. At the core the question is how can remote servers make sure that you are entitled to remove the content you are asking them to remove. The main way to do this is by checking the cryptographic key associated with your account. But if you're trying to do this on someone else's content, there is no match since the author account is different from the deletor account.

This is why protocols like Diaspora have a concept of ownership. Yes, you're the author of your content, but if you post on someone else's post, you're effectively transferring ownership to them, and from there they can remove it (but not edit it, since they aren't the author).

However I don't think the same concept applies to ActivityPub, so we can't ensure that a comment deleted on your post will be deleted globally.

**Comment by annando** [comment]
I hadn't looked in detail into this PR. Just a short note: Via the Diaspora protocol we can delete comments from other people on our own posts and this will be done globally. With AP this is not so easy. This could/should work with deletions on Friendica systems but not with Mastodon. They handle these deletion requests differently. While we say that the thread starter is a kind of "owner" of all the posts to this thread, this is not a common believe in the fediverse.

**Comment by MrPetovan** [comment]
I should add before it becomes a tangent of its own that this issue is more about blocked contacts' ability to view or comment at all in the first place, not about deleting.
